src/roberta/roberta_dataset_builder.py

```diff
@@ -0,0 +1,69 @@
+from typing import Dict, List, Union
+
+import datasets
+import transformers
+
+from ..dataset_builder import DatasetBuilder
+from ..utils import list_dict_to_dict_list
+
+
+class RobertaDatasetBuilder(DatasetBuilder):
+    """Roberta DatasetBuilder"""
+
+    def __init__(
+        self,
+        tokenizer: transformers.PreTrainedTokenizerFast,
+        max_length: int,
+        key: str,
+        reset_position_ids=False,
+    ) -> None:
+        self.features = datasets.Features(
+            {
+                "input_ids": datasets.Sequence(datasets.Value("int32")),
+                "attention_mask": datasets.Sequence(datasets.Value("uint8")),
+                "position_ids": datasets.Sequence(datasets.Value("int32")),
+                "special_tokens_mask": datasets.Sequence(datasets.Value("uint8")),
+            }
+        )
+
+        self.tokenizer = tokenizer
+        self.max_length = max_length
+        self.key = key
+        self.reset_position_ids = reset_position_ids
+        self.buffer = []
+
+    def make_examples(self, batch: Dict) -> Dict:
+        examples = []
+        docs = self.tokenizer(
+            batch[self.key], padding=False, add_special_tokens=False, truncation=False, return_attention_mask=False
+        )["input_ids"]
+
+        for doc in docs:
+            doc += [self.tokenizer.sep_token_id]
+            ex = self._add_document(doc)
+            if ex:
+                examples.append(ex)
+        return list_dict_to_dict_list(examples)
+
+    def _add_document(self, input_ids: List[int]) -> Union[transformers.BatchEncoding, None]:
+        self.buffer += input_ids
+        if len(self.buffer) >= self.max_length - 1:
+            ex = self._create_example()
+            return ex
+        return None
+
+    def _create_example(self) -> transformers.BatchEncoding:
+        chunk = self.buffer[: self.max_length - 1]
+        chunk = [self.tokenizer.cls_token_id] + chunk
```

Comment:
```suggestion
        chunk.insert(0, self.tokenizer.cls_token_id)
```
로 하는 것과 연산 차이가 없을까요?

Reply:
기존 방식은 리스트를 각각 만들어서 합치는 방식이라 더 개선 될 것 같아요.
실험도 그렇구요

```
import time

def main():
    test_vec = list(range(0, 1000000))
    s = time.time()
    test_vec = [-1] + test_vec
    print(time.time() - s)

    test_vec = list(range(0, 1000000))
    s = time.time()
    test_vec.insert(0, -1)
    print(time.time() - s)


if __name__ == '__main__':
    main()
```